src/stan/math/prim/scal/prob/neg_binomial_2_log_rng.hpp

```diff
@@ -31,20 +31,45 @@ namespace stan {
                            RNG& rng) {
       using boost::variate_generator;
       using boost::random::negative_binomial_distribution;
+      using boost::random::poisson_distribution;
+      using boost::gamma_distribution;
 
       static const char* function("stan::math::neg_binomial_2_log_rng");
+      static const double POISSON_MAX_RATE = std::pow(2.0, 30);
 
       using stan::math::check_finite;
       using stan::math::check_positive_finite;
+      using stan::math::check_not_nan;
+      using stan::math::check_nonnegative;
+      using stan::math::check_less;
 
       check_finite(function, "Log-location parameter", eta);
       check_positive_finite(function, "Precision parameter", phi);
 
+      double exp_eta_div_phi = std::exp(eta)/phi;
```

Comment:
Because we are calling boost::gamma_distribution directly instead of stan::math::gamma_rng. See file src/stan/math/prim/scal/prob/gamma_rng.hpp:


      /*
        the boost gamma distribution is defined by
        shape and scale, whereas the stan one is defined
        by shape and rate
      */
      variate_generator<RNG&, gamma_distribution<> >
        gamma_rng(rng, gamma_distribution<>(alpha, 1.0 / beta));
      return gamma_rng();